core/tests/data/fs/file_system.cpp

```diff
@@ -316,6 +326,10 @@ TEST_CASE("data::FileSystem") // NOLINT(readability-function-size)
         }
         else
         {
+            // Test copying files
+            REQUIRE_FALSE(FileSystem::copy("/bar/nonexistent", "/bar/newfiledoesntmatter"));
```

Comment:
Would also be nice to cover cases such as copying to somewhere read-only (e.g., outside the archive).
```suggestion
            REQUIRE_FALSE(FileSystem::copy("/dir/sub/arc/bar/nonexistent", "/dir/sub/arc/bar/newfiledoesntmatter"));
            REQUIRE_FALSE(FileSystem::copy(baz->path(), "/dir/readonlyfile"));
```